src/vm.rs

```diff
@@ -414,34 +454,75 @@ impl<'a, S: Storage, C: PevmChain> Database for VmDb<'a, S, C> {
     }
 
     fn storage(&mut self, address: Address, index: U256) -> Result<U256, Self::Error> {
+        if let LazyStrategy::ERC20Transfer {
+            sender_balance_slot,
+            recipient_balance_slot,
+        } = self.lazy_strategy
+        {
+            if index == sender_balance_slot {
+                return self
+                    .vm
+                    .storage
+                    .storage(&address, &index)
+                    .map_err(|err| ReadError::StorageError(err.to_string()));
+            }
+            if index == recipient_balance_slot {
+                return self
+                    .vm
+                    .storage
+                    .storage(&address, &index)
+                    .map_err(|err| ReadError::StorageError(err.to_string()));
+            }
+        }
+
         let location_hash = self
             .vm
             .hasher
             .hash_one(MemoryLocation::Storage(address, index));
 
         let read_origins = self.read_set.entry(location_hash).or_default();
+        read_origins.clear();
+
+        let mut accumulated_addition = U256::ZERO;
```

Comment:
Hmm, ideally we would put these logic in the lazy recipient branch above since unlike balance & nonce the operations on storage slots can be arbitrary (not addition in all cases).

Reply:
added `let mut accumulated_subtraction = U256::ZERO; `